src/flint/fmpz.jl

```diff
@@ -2670,58 +2650,52 @@ end
 add!(z::ZZRingElem, a::ZZRingElem, b::Integer) = add!(z, a, ZZRingElem(b))
 add!(z::ZZRingElem, x::Int, y::ZZRingElem) = add!(z, y, x)
 
-function neg!(z::ZZRingElem, a::ZZRingElem)
+function neg!(z::ZZRingElemOrPtr, a::ZZRingElemOrPtr)
   ccall((:fmpz_neg, libflint), Nothing,
         (Ref{ZZRingElem}, Ref{ZZRingElem}),
         z, a)
   return z
 end
 
-function neg!(a::ZZRingElem)
-  return neg!(a,a)
-end
+neg!(a::ZZRingElemOrPtr) = neg!(a, a)
 
-function sub!(z::ZZRingElem, a::ZZRingElem, b::ZZRingElem)
+function sub!(z::ZZRingElemOrPtr, a::ZZRingElemOrPtr, b::ZZRingElemOrPtr)
   ccall((:fmpz_sub, libflint), Nothing,
         (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}),
         z, a, b)
   return z
 end
 
-function sub!(z::ZZRingElem, a::ZZRingElem, b::Int)
+function sub!(z::ZZRingElemOrPtr, a::ZZRingElemOrPtr, b::Int)
   ccall((:fmpz_sub_si, libflint), Nothing,
         (Ref{ZZRingElem}, Ref{ZZRingElem}, Int),
         z, a, b)
   return z
 end
 
-function sub!(z::ZZRingElem, a::ZZRingElem, b::UInt)
+function sub!(z::ZZRingElemOrPtr, a::ZZRingElemOrPtr, b::UInt)
   ccall((:fmpz_sub_ui, libflint), Nothing,
         (Ref{ZZRingElem}, Ref{ZZRingElem}, UInt),
         z, a, b)
   return z
 end
 
-function sub!(z::ZZRingElem, a::ZZRingElem, b::Integer)
+function sub!(z::ZZRingElemOrPtr, a::ZZRingElemOrPtr, b::Integer)
   return sub!(z, a, ZZRingElem(b))
 end
 
-function sub!(z::ZZRingElem, b::Integer, a::ZZRingElem)
+function sub!(z::ZZRingElemOrPtr, b::Integer, a::ZZRingElemOrPtr)
   sub!(z, a, b)
   return neg!(z, z)
 end
 
 
-function mul!(z::ZZRingElem, x::ZZRingElem, y::ZZRingElem)
+function mul!(z::ZZRingElemOrPtr, x::ZZRingElemOrPtr, y::ZZRingElemOrPtr)
   ccall((:fmpz_mul, libflint), Nothing,
         (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), z, x, y)
   return z
 end
 
-function mul!(a::Ref{ZZRingElem}, b::Ref{ZZRingElem}, c::ZZRingElem)
-  ccall((:fmpz_mul, libflint), Nothing, (Ref{ZZRingElem}, Ref{ZZRingElem}, Ref{ZZRingElem}), a, b, c)
-end
```

Comment:
This method is now subsumed into the preceding one, thanks to the `ZZRingElemOrPtr` union.